File: src/wifi_framework/capabilities.py
from dataclasses import dataclass


@dataclass(frozen=True)
class WifiCapabilities:
    bands_2ghz: bool = False
    bands_5ghz: bool = False
    bands_6ghz: bool = False
    ap: bool = False
    sta: bool = False
    concurrent_ap_sta: bool = False
    mesh_80211s: bool = False
    ieee80211k: bool = False
    ieee80211v: bool = False
    ieee80211r: bool = False


def supported_features(cap: WifiCapabilities) -> list[str]:
    return [name for name, enabled in vars(cap).items() if enabled]

File: src/core/capabilities.ts
export interface WifiCapabilities {
  bands_2ghz: boolean;
  bands_5ghz: boolean;
  bands_6ghz: boolean;
  ap: boolean;
  sta: boolean;
  concurrent_ap_sta: boolean;
  mesh_80211s: boolean;
  ieee80211k: boolean;
  ieee80211v: boolean;
  ieee80211r: boolean;
}

export const defaultCapabilities: WifiCapabilities = {
  bands_2ghz: true,
  bands_5ghz: true,
  bands_6ghz: false,
  ap: true,
  sta: true,
  concurrent_ap_sta: true,
  mesh_80211s: true,
  ieee80211k: true,
  ieee80211v: true,
  ieee80211r: false,
};

export function supportedFeatures(cap: WifiCapabilities): string[] {
  return Object.entries(cap)
    .filter(([_, enabled]) => Boolean(enabled))
    .map(([name]) => name);
}
